bytecomp/bytegen.ml

```diff
@@ -223,8 +231,37 @@ let rec size_of_lambda env = function
   | Lprim (Pduprecord (Record_extension _, size), _, _) ->
       RHS_block (size + 1)
   | Lprim (Pduprecord (Record_float, size), _, _) -> RHS_floatblock size
+  | Lprim (Praise _, _, _) -> RHS_unreachable
   | Levent (lam, _) -> size_of_lambda env lam
   | Lsequence (_lam, lam') -> size_of_lambda env lam'
+  | Lifthenelse (_cond, ifso, ifnot) ->
+      let size_ifso = size_of_lambda env ifso in
+      let size_ifnot = size_of_lambda env ifnot in
+      join_rhs_kind size_ifso size_ifnot
+  | Lstaticraise (_, _) -> RHS_unreachable
+  | Lstaticcatch (body, _params, handler) ->
+      (* If needed, we could refine the sizes of the parameters by propagating
+         the sizes from the Lstaticraise sites. *)
+      let size_body = size_of_lambda env body in
+      let size_handler = size_of_lambda env handler in
+      join_rhs_kind size_body size_handler
+  | Lswitch (_arg, sw, _loc) ->
+      List.fold_left (fun acc (_const, act) ->
+          join_rhs_kind acc (size_of_lambda env act))
+        (List.fold_left (fun acc (_tag, act) ->
+             join_rhs_kind acc (size_of_lambda env act))
+           (match sw.sw_failaction with
+            | None -> RHS_unreachable
+            | Some act -> size_of_lambda env act)
+           sw.sw_blocks)
+        sw.sw_consts
+  | Lstringswitch (_arg, cases, default, _loc) ->
+      List.fold_left (fun acc (_string, act) ->
+          join_rhs_kind acc (size_of_lambda env act))
+        (match default with
+         | None -> RHS_unreachable
+         | Some act -> size_of_lambda env act)
+        cases
   | _ -> RHS_nonrec
```

Comment:
should we maybe consider a non-fragile pattern here to see what we are missing out?

Reply:
I agree with you, but it takes a lot of time to understand if any given expression is safe or not so I thought submitting a PR with straightforward fixes for a known, reported bug should take precedence.
You can find some hints of why things are complicated in the very long PR description.